library/core/src/primitive_docs.rs

```diff
@@ -1387,9 +1387,19 @@ mod prim_usize {}
 /// returning values from safe functions; such violations may result in undefined behavior. Where
 /// exceptions to this latter requirement exist, they will be called out explicitly in documentation.
 ///
+/// For references to [slices](primitive.slice.html) and [`str`s](primitive.str.html),
+/// a consequence of the above is that their lengths must always be short enough that
+/// `size_of_val(t) <= isize::MAX`. Said otherwise, for an element type `E` where
+/// `size_of::<E>() > 0` (a non-ZST), the length of the slice must never exceed
+/// `isize::MAX / size_of::<E>()`. (Raw pointers may have longer lengths, but
+/// references must not.  For example, compare the documentation of
+/// [`ptr::slice_from_raw_parts`](ptr/fn.slice_from_raw_parts.html) and
+/// [`slice::from_raw_parts`](slice/fn.from_raw_parts.html).)
```

Comment:
Is this meant to apply to all types with slice tails? In `&(i32, [i32])`, is the max length reduced by one to make sure that the entire type always fits into `isize`?